conformance testdata: remove tags and trim spaces

yujuhong opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
4. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What this PR does / why we need it**:
To detect whether conformance tests have been changed, the test
argument with the `ConformanceIt` is recorded to conformance.txt.
This change remove additional tags (e.g., [Feature:]) and trim spaces,
so that the detection is less-prone to noise.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
NONE
```

yujuhong committed:
conformance: normalize the test names

To detect whether conformance tests have been changed, the test
argument with the `ConformanceIt` is recorded to conformance.txt.
This change remove additional tags (e.g., [Feature:]) and trim spaces,
so that the detection is less-prone to noise.

This change also updates conformance.txt with normalized names.

yujuhong commented:
Background: I'd like to add tags to some of the conformance tests without triggering a change in the conformance test data.

/cc @dchen1107

mml commented on test/conformance/walk.go:
```diff
@@ -197,6 +198,19 @@ func (v *visitor) getDescription(value string) string {
+	regexTag = regexp.MustCompile(`(\[[a-zA-Z0-9:-]+\])`)
+)
+
+func processValue(s string) string {
```
Maybe call this `normalizeTestName`?

yujuhong replied:
Done.

mml commented:
As long as no objections on the conformance list, this LG otherwise.  Thanks!

yujuhong commented:
> As long as no objections on the conformance list, this LG otherwise. Thanks!

Just sent out the email. Will wait for folks to respond. Thanks for reviewing!

/cc @bgrant0607 @smarterclayton for `testdata` changes

yujuhong commented:
/retest

mml commented:
/approve

yujuhong commented:
FYI, email to the cncf-kubernetes-conformance mailing list: https://lists.cncf.io/g/cncf-k8s-conformance/message/414

bgrant0607 commented:
@yujuhong The current impact of this looks like it's just removing trailing spaces?

Are there any automated consumers of conformance.txt that this might impact?

yujuhong commented:
> @yujuhong The current impact of this looks like it's just removing trailing spaces?

It also removed the `[Flaky]` and `[Slow]` tags. I uploaded an incomplete version before. Updated the PR. 

> Are there any automated consumers of conformance.txt that this might impact?

I don't know who are the consumers of this file. That's part of the reason why I sent out the email to the mailing list. Could you point me to any existing consumers? Thanks!

duglin commented:
ping @brahmaroutu

yujuhong commented:
/retest

bgrant0607 commented:
cc @jagosan

bgrant0607 commented:
Flaky tests shouldn't be part of conformance. #62161

bgrant0607 commented:
Does https://github.com/kubernetes/kubernetes/blob/master/test/conformance/conformance_test.sh run as presubmit?

Does anything else consume conformance.txt?

cc @timothysc

bgrant0607 commented:
Otherwise the change looks fine to me. Thanks.

yujuhong commented:
Added a unit test for the function. Nothing else was changed.

> Does https://github.com/kubernetes/kubernetes/blob/master/test/conformance/conformance_test.sh run as presubmit?

Yes, that's how I caught the issue.

yujuhong commented:
> Does anything else consume conformance.txt?

FWIW, a simple search in https://github.com/heptio/sonobuoy returned nothing.

bgrant0607 commented:
/lgtm
/approve
